components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -101,12 +101,16 @@ func installWorkspaceRoutes(r *mux.Router, config *RouteHandlerConfig, ip Worksp
 	routes.HandleDirectSupervisorRoute(r.PathPrefix("/_supervisor/v1"), true)
 	routes.HandleDirectSupervisorRoute(r.PathPrefix("/_supervisor"), true)
 
-	routes.HandleDirectIDERoute(enableCompression(r).MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
-		// this handles all foreign (none-IDE) content
-		return m.Vars != nil && m.Vars[foreignOriginIdentifier] != ""
-	}))
-
-	routes.HandleRoot(enableCompression(r).NewRoute())
+	rootRouter := enableCompression(r)
+	rootRouter.Use(func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+			if req.URL.Path == "/vscode-extension-auth-callback" {
```

Comment:
what was an issue with using callback.html? The PR description does not explain anything.

Reply:
We can use `callback` (not sure about callback.html it should be an endpoint not a resource) but I thought that's too generic, maybe in the future we add another web ide that has some type of callback too.

Reply:
Just to understand, we would like to remove cross checks and auth for auth callback? I do not yet understand what this PR is fixing.